libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -857,13 +855,12 @@ func setPerPortForwarding(b portBinding, ipv iptables.IPVersion, bridgeName stri
 		// to fill the checksum.
 		//
 		// https://github.com/torvalds/linux/commit/c80fafbbb59ef9924962f83aac85531039395b18
-		args = []string{
+		rule := iptables.Rule{IPVer: ipv, Table: iptables.Mangle, Chain: "POSTROUTING", Args: []string{
 			"-p", b.Proto.String(),
 			"--sport", strconv.Itoa(int(b.Port)),
 			"-j", "CHECKSUM",
 			"--checksum-fill",
-		}
-		rule := iptRule{ipv: ipv, table: iptables.Mangle, chain: "POSTROUTING", args: args}
+		}}
 		if err := appendOrDelChainRule(rule, "SCTP CHECKSUM", enable); err != nil {
```

Comment:
If we're moving, would it make sense to put the `comment` as a field in the Rule? Possibly the error-message generating could then be fully delegated to the `rule.Append` and `rule.Delete` functions, because they both know whether something was added or deleted that failed, and they would have the comment related to it;
https://github.com/moby/moby/blob/40f58b3492f97d6afa5603781daa8c4002a957cd/libnetwork/drivers/bridge/setup_ip_tables_linux.go#L544-L555

Reply:
Good idea!

I originally thought about adding a `Comment` field to automatically add `-m comment --comment <COMMENT>` to iptables rules, but that's quite involved because we need to handle 'migration' (eg. we should not create duplicate rules, one with and one without a comment). But what you're proposing sounds good too.

However, that will need a bit more work as not all rules are currently inserted by an helper that takes a comment, and some comments might be too unspecific to be really helpful when debugging.

Since this PR doesn't deal with comments, I'd prefer addressing that in a follow-up.